api/krusty/namespaces_test.go

```diff
@@ -801,3 +801,31 @@ metadata:
   namespace: iter8-monitoring
 `)
 }
+
+// Demonstrates that metadata.name is only overridden for a kind: Namespace with apiVersion: v1
+// Test for issue #5072
+func TestNameNotOveriddenForNonCoreApiVersionOnANamespaceKind(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+
+	th.WriteF("azure-servicebus.yaml", `
+apiVersion: servicebus.azure.com/v1beta20210101preview
+kind: Namespace
+metadata:
+  name: core-sb-99
+  namespace: podinfo
```

Comment:
Please change the namespace name to without `podinfo`. 
I think this test case can't check to change `metadata.namespace`.

Reply:
done